Donation-platform/src/main/java/com/google/RecommendationSystem.java

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google;
+
+import com.google.Charity;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Arrays;
+
+/* Class that takes user-selected tags as input and finds the according best-matching charities.*/
+public class RecommendationSystem {
```

Comment:
Recommendation is a generic word, probably use GetPersonalizedRecommendations.java

Reply:
I named it RecommendationSystem.java to reflect that it is a version of a recommendation system (which seems to be what other developers/researchers/companies call this type of algorithm). I agree that in our case something like "PersonalizedRecommendations.java" is more descriptive of our specific use case, so I'll change it to that.

Reply:
Fixed with recent commit.